### PR TITLE
Contract/Donation Services

### PR DESCRIPTION
**Pull request by m1sterc001guy**
This CR adds services that persist contract and donation data into the database.

It also adds some dummy scripts to create dummy data in the database.

### CODE REVIEW

File: api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractModel.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.UUID;
+
+@Entity
+@Table(name = "contracts")
+@Builder
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
+@Setter(value = AccessLevel.PACKAGE)
+@Getter
+public class ContractModel {
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractModel.java]: Would we add the paypalPaymentId to this class or does that go someplace else? Maybe a paymentsModel makes sense? Im not sure where the place to stick that is yet, maybe we could discuss that on wednesday

File: api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.UUID;
+import java.sql.Timestamp;
+
+@RequiredArgsConstructor
+public class ContractServiceImpl implements ContractService {
+
+    @NonNull final ContractModelRepository contractModelRepository;
+
+    @Override
+    public ContractModel createContract(@NonNull UUID proposerId, @NonNull UUID streamerId, @Nullable String game, @NonNull String description) {
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java]: nit: final on method params that dont change. applies to the cr as a whole

File: api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        final Timestamp creationTimestamp = new Timestamp(System.currentTimeMillis());
+        // Add 10 days to the current timestamp to create the expires timestamp
+        final Calendar cal = Calendar.getInstance();
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java]: super-super-optional: Im a fan of spelling out the word. I personally would rather read "calendar" than a shortened "cal"... I realize this is my personal preference and not something ill force

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java [hunk 4]
```diff
@@ -19,12 +25,30 @@
...
         final Payment payment = paymentsService.executePayment(createDonationRequest.getPayPalPaymentId()).orElse(null);
         if(payment == null) {
             return "couldn't execute payment for id: " + createDonationRequest.getPayPalPaymentId();
         }
+
+        final String streamerName = "CrazyHorse";
+        final Optional<UserModel> proposerObj = userService.getUser(createDonationRequest.getUsername());
+        final Optional<UserModel> streamerObj = userService.getUser(streamerName);
+        if (!proposerObj.isPresent() || !streamerObj.isPresent()) {
+            return "users do not exist";
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java]: This should return HTTP.InvalidRequest or something like that i believe. Additionally this feels like a good place to log something like ->

```
log.warn(Attempted to create donation for non existant user: + userNameThatRetunedNull)
```

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import java.sql.Timestamp;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+public class DonationServiceImpl implements  DonationService{
+
+    @NonNull final DonationModelRepository donationModelRepository;
+
+    @Override
+    public void createDonation(UUID contractId, UUID donatorId, BigDecimal donationAmount, Timestamp contractTimestamp) {
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationServiceImpl.java]: nit: final and @nullable or @Nonnull on method params

File: api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractModel.java (hunk 1 above)
Comment (m1sterc001guy) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractModel.java]: We should add it to the donation table. I'll make the change

File: api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java (hunk 3 above)
Comment (m1sterc001guy) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractServiceImpl.java]: Think I was copying this from stack overflow or something, I'm a fan of writing it out as well.

**Pull request merged**